classes/task/fetch_recording_metadata_task.php

```diff
@@ -54,15 +54,62 @@ public function execute() {
         require_once(__DIR__.'/../../locallib.php');
         global $CFG, $DB;
 
+        // Updates older records with no recordid in meta or log record, to have
+        // them where possible.
+        $sql = "SELECT id, meetingid, meta, timecreated, recordid
+                  FROM {bigbluebuttonbn_logs}
+                 WHERE recordid IS NULL
+                   AND log = :log
+                   AND NOT ".$DB->sql_like('meta', ':recordedfalse', false, false, $notlike = true)."
+                   AND ".$DB->sql_like('meta', ':recordtrue');
+        $createlogswithnorecordid = $DB->get_records_sql($sql, [
+            'log' => BIGBLUEBUTTONBN_LOG_EVENT_CREATE,
+            'recordtrue' => '%"record":"true"%',
+            'recordedfalse' => '%"recorded":false%', // Recording did not happen (set after meeting ended).
+        ], 0, $numberofrecords = 100);
+
+        $foundMatching = 0;
+        $notMatching = 0;
+        foreach ($createlogswithnorecordid as $logentry) {
+            $meta = json_decode($logentry->meta);
+            $recordings = bigbluebuttonbn_get_recordings_array_fetch_page([$logentry->meetingid]);
+            $chosenrecordid = null;
+            $closest = null;
+            foreach ($recordings as $recordid => $recordinginfo) {
+                $starttime = $recordinginfo['startTime'] / 1000;
+                // Validate the startTime of the recording is within the same time as the log entry (+-3 seconds)
+                if (abs($logentry->timecreated - $starttime) > 3) {
+                    continue;
+                }
+                if ($closest === null || abs($logentry->timecreated - $closest) > abs($starttime - $logentry->timecreated)) {
```

Comment:
A comment explaining the logic to explain the matching would be nice for someone trying to grok the code. its a little opaque from just the code atm

Reply:
Sure thing